tests/test_evaluation.py

```diff
@@ -0,0 +1,141 @@
+    population_size=1e4,
+    recombination_rate=1e-8,
+    random_seed=1024,
+)
+true_ts = msprime.sim_mutations(true_ts, rate=2e-8, random_seed=1024)
+smpdata = tsinfer.SampleData.from_tree_sequence(true_ts)
+infr_ts = tsinfer.infer(smpdata).simplify(filter_sites=False)
+
+
+def naive_shared_node_spans(ts, other):
+    """
+    Inefficient but transparent function to get span where nodes from two tree
+    sequences subtend the same sample set
+    """
+
+    def _clade_dict(tree):
+        clade_to_node = defaultdict(set)
+        for node in tree.nodes():
+            clade = frozenset(tree.samples(node))
+            clade_to_node[clade].add(node)
+        return clade_to_node
+
+    assert ts.sequence_length == other.sequence_length
+    assert ts.num_samples == other.num_samples
+    query_tree = ts.first()
+    target_tree = other.first()
+    query = _clade_dict(query_tree)
+    target = _clade_dict(target_tree)
+    out = np.zeros((ts.num_nodes, other.num_nodes))
+    right = 0
+    while right != ts.sequence_length:
+        left = right
+        right = min(query_tree.interval[1], target_tree.interval[1])
+        for clade, nodes in query.items():
+            if clade in target:
+                for i in nodes:
+                    for j in target[clade]:
+                        out[i, j] += right - left
+        if query_tree.interval[1] == right:
+            query_tree.next()
+            query = _clade_dict(query_tree)
+        if target_tree.interval[1] == right:
+            target_tree.next()
+            target = _clade_dict(target_tree)
+    return scipy.sparse.csr_matrix(out)
+
+
+@pytest.mark.parametrize("ts", [true_ts, infr_ts])
+class TestCladeMap:
+    def test_map(self, ts):
+        """
+        test that clade map has correct nodes, clades
+        """
+        clade_map = evaluation.CladeMap(ts)
+        for tree in ts.trees():
+            for node in tree.nodes():
+                clade = frozenset(tree.samples(node))
+                assert node in clade_map._nodes[clade]
+                assert clade_map._clades[node] == clade
+            clade_map.next()
+
+    def test_diff(self, ts):
+        """
+        test difference in clades between adjacent trees
+        """
+        clade_map = evaluation.CladeMap(ts)
+        tree_1 = ts.first()
+        tree_2 = ts.first()
+        while True:
+            tree_2.next()
+            diff = clade_map.next()
+            diff_test = {}
+            for n in set(tree_1.nodes()) | set(tree_2.nodes()):
+                prev = frozenset(tree_1.samples(n))
+                curr = frozenset(tree_2.samples(n))
+                if prev != curr:
+                    diff_test[n] = (prev, curr)
+            for node in diff_test.keys() | diff.keys():
+                assert diff_test[node][0] == diff[node][0]
+                assert diff_test[node][1] == diff[node][1]
+            if tree_2.index == ts.num_trees - 1:
+                break
+            tree_1.next()
```

Comment:
It would be good to have a test example where more than one node has the same set of subtended samples? (I think an msprime.CensusEvent would produce this?)